From 0c4c580bb3a402ff5f937209ffb5abd2cb6e0b7e Mon Sep 17 00:00:00 2001
From: Austin Gill <notgnoshi@gmail.com>
Date: Thu, 12 Sep 2019 19:09:19 -0600
Subject: [PATCH] Fix generated PDF bookmark heirarchy.

* When resetting \thechapter to start numbering the appendices, also reset
  \theHchapter.
* Prefix every \addcontentsline with \cleardoublepage\phantomsection

See
https://tex.stackexchange.com/questions/57284/pdf-contains-incorrect-bookmark-hierarchy,
https://tex.stackexchange.com/questions/23499/incorrect-bookmarks-and-page-number-in-table-of-contents,
https://tex.stackexchange.com/questions/44088/when-do-i-need-to-invoke-phantomsection
---
 thesis.cls | 66 ++++++++++++++++++++++++------------------------
 thesis.dtx | 74 +++++++++++++++++++++++++++---------------------------
 2 files changed, 70 insertions(+), 70 deletions(-)

diff --git a/thesis.cls b/thesis.cls
index beae270..a124614 100644
--- a/thesis.cls
+++ b/thesis.cls
@@ -125,7 +125,7 @@
 \newcommand{\body}{\doublespace\vfill\pagebreak\eject
   \pagenumbering{arabic}\setcounter{page}{1}\eject}
 \newcommand{\supplementaries}{\renewcommand{\@chapapp}{Appendix}
-\setcounter{chapter}{0}\renewcommand{\thechapter}{\Alph{chapter}}}
+\setcounter{chapter}{0}\renewcommand{\theHchapter}{\Alph{chapter}}\renewcommand{\thechapter}{\Alph{chapter}}}
 \renewcommand{\maketitle}{%
 \null%
 \singlespace%
@@ -225,67 +225,67 @@ Date Defended: \Zdefdate\\%
   }}
 %%\renewenvironment{abstract}{\singlespace\newpage
 \renewenvironment{abstract}{\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}Abstract}
-  \addcontentsline{toc}{chapter}{Abstract}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Abstract}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Abstract}
   \@makeschapterhead{\centerline{Abstract}}
   \vskip 20pt\par\singlespace}{\newpage}%
 %% This version prints the thesis/dissertation title
 %% \renewenvironment{abstract}{\singlespace
-%%   \addcontentsline{toc}{chapter}{Abstract}
+%%   \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Abstract}
 %%   \@makeschapterhead{\centerline{Abstract}}
 %%   \vskip 20pt\par\noindent Title: \Ztitle\par
 %%   \vskip 0.5\baselineskip\strut\\%
 %%   \noindent}{\newpage}%
 %% \newenvironment{acknowledgments}{\singlespace\newpage
 \newenvironment{acknowledgments}{\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}Acknowledgments}
-  \addcontentsline{toc}{chapter}{Acknowledgments}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Acknowledgments}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Acknowledgments}
   \@makeschapterhead{\centerline{Acknowledgments}}
   \vskip 20pt\par}{\newpage}
 \def\contentsname{Table of Contents}
 \renewcommand{\tableofcontents}{\singlespace\newpage
-  \addcontentsline{toc}{chapter}{Table of Contents}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Table of Contents}
   \@makeschapterhead{\centerline{Table of Contents}}
   \vskip 20pt\par\@starttoc{toc}}
 \renewcommand{\listoftables}{\singlespace\newpage
-\addcontentsline{toc}{chapter}{List of Tables}
-%%\addcontentsline{toc}{chapter}{\protect\numberline{}List of Tables}
+\cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Tables}
+%%\cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Tables}
   \@makeschapterhead{\centerline{List of Tables}}
   \vskip20pt
   \@starttoc{lot}}
 \renewcommand{\listoffigures}{\singlespace\newpage
-  \addcontentsline{toc}{chapter}{List of Figures}
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}List of Figures}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Figures}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Figures}
   \@makeschapterhead{\centerline{List of Figures}}
   \vskip20pt
   \@starttoc{lof}}
 \newenvironment{listofsymbols}{\singlespace\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}List of Symbols and Acronyms}
-  \addcontentsline{toc}{chapter}{List of Symbols and Acronyms}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Symbols and Acronyms}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Symbols and Acronyms}
   \@makeschapterhead{\centerline{List of Symbols and Acronyms}}
   \vskip 20pt\par}{\newpage}
 \newenvironment{listofkeywords}{\singlespace\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}List of Keywords}
-  \addcontentsline{toc}{chapter}{List of Keywords}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Keywords}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Keywords}
   \@makeschapterhead{\centerline{List of Keywords}}
   \vskip 20pt\par}{\newpage}
 %%\newenvironment{dedication}{\singlespace\newpage
 \newenvironment{dedication}{\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}Dedication}
-  \addcontentsline{toc}{chapter}{Dedication}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Dedication}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Dedication}
   \@makeschapterhead{\centerline{Dedication}}
   \vskip 20pt\par}{\newpage}
 %%\newenvironment{preface}{\singlespace\newpage
 \newenvironment{preface}{\newpage
-  %%  \addcontentsline{toc}{chapter}{\protect\numberline{}Preface}
-  \addcontentsline{toc}{chapter}{Preface}
+  %%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Preface}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Preface}
   \@makeschapterhead{\centerline{Preface}}
   \vskip 20pt\par}{\newpage}
 \renewcommand{\thebibliography}[1]{
  \@makeschapterhead{\centerline{Bibliography}}
  \vskip20pt
-%% \addcontentsline{toc}{chapter}{\protect\numberline{}References}
- \addcontentsline{toc}{chapter}{Bibliography}
+%% \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}References}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Bibliography}
  \singlespace
  \list{[\arabic{enumi}]}{\settowidth\labelwidth{[#1]}
  \leftmargin\labelwidth
@@ -300,7 +300,7 @@ Date Defended: \Zdefdate\\%
  \sfcode `\.=1000\relax}
 \newenvironment{appendices}{
 \if@openright\cleardoublepage\else\clearpage\fi
- \addcontentsline{toc}{chapter}{Appendices}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Appendices}
  \thispagestyle{plain}\parindent\z@
  \parskip\z@ \@plus .3\p@\relax}
   {\clearpage}
@@ -312,12 +312,12 @@ Date Defended: \Zdefdate\\%
 \def\@appendix[#1]#2{\ifnum \c@secnumdepth >\m@ne
                          \refstepcounter{chapter}%
                          \typeout{\@chapapp\space\thechapter.}%
-                         \addcontentsline{toc}{section}%
+                         \cleardoublepage\phantomsection\addcontentsline{toc}{section}%
                          %%{\protect\numberline{Appendix \thechapter}}%
                          {Appendix \thechapter}%
                     \else
-                    %% \addcontentsline{toc}{section}{#1}%
-                    %% \addcontentsline{toc}{section}{}%
+                    %% \cleardoublepage\phantomsection\addcontentsline{toc}{section}{#1}%
+                    %% \cleardoublepage\phantomsection\addcontentsline{toc}{section}{}%
                     \fi
                     \chaptermark{#1}%
                     \addtocontents{lof}{\protect\addvspace{10\p@}}%
@@ -330,15 +330,15 @@ Date Defended: \Zdefdate\\%
                     \fi}
 \newenvironment{gloss}{
  \newpage\@makeschapterhead{\centerline{Glossary of Terms}}
-%% \addcontentsline{toc}{chapter}{\protect\numberline{}Glossary of Terms}
- \addcontentsline{toc}{chapter}{Glossary of Terms}
+%% \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Glossary of Terms}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Glossary of Terms}
  \thispagestyle{plain}\parindent\z@
  \parskip\z@ \@plus .3\p@\relax}
  {\clearpage}
 \newenvironment{abbreviations}{
  \@makeschapterhead{\centerline{List of Abbreviations}}
- %% \addcontentsline{toc}{chapter}{\protect\numberline{}List of Abbreviations}
- \addcontentsline{toc}{chapter}{List of Abbreviations}
+ %% \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Abbreviations}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Abbreviations}
  \thispagestyle{plain}\parindent\z@
  \parskip\z@ \@plus .3\p@\relax}
  {\clearpage}
@@ -347,15 +347,15 @@ Date Defended: \Zdefdate\\%
  \columnseprule \z@
  \columnsep 35\p@
  \twocolumn[\@makeschapterhead{\centerline{\indexname}}]%
-%% \addcontentsline{toc}{chapter}{\protect\numberline{}Index}
- \addcontentsline{toc}{chapter}{Index}
+%% \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Index}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Index}
  \thispagestyle{plain}\parindent\z@
  \parskip\z@ \@plus .3\p@\relax
  \let\item\@idxitem}
  {\onecolumn\clearpage}
 \newenvironment{vita}{\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}Vita}
-  \addcontentsline{toc}{chapter}{Vita}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Vita}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Vita}
   \@makeschapterhead{\centerline{Vita}}
   \vskip 20pt\par}{\newpage}
 \newcommand{\singlespace}{\@normalsize \baselineskip \normalbaselineskip}
diff --git a/thesis.dtx b/thesis.dtx
index 7a3608c..b2befa1 100644
--- a/thesis.dtx
+++ b/thesis.dtx
@@ -309,7 +309,7 @@
 % bibliography, appendices, glossary, index, and vita.  
 % \iffalse
 \newcommand{\supplementaries}{\renewcommand{\@chapapp}{Appendix}
-\setcounter{chapter}{0}\renewcommand{\thechapter}{\Alph{chapter}}}
+\setcounter{chapter}{0}\renewcommand{\theHchapter}{\Alph{chapter}}\renewcommand{\thechapter}{\Alph{chapter}}}
 % \fi
 %
 %\section{Commands and Environments in the Preliminaries}
@@ -433,13 +433,13 @@ Date Defended: \Zdefdate\\%
 % \iffalse
 %%\renewenvironment{abstract}{\singlespace\newpage
 \renewenvironment{abstract}{\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}Abstract}
-  \addcontentsline{toc}{chapter}{Abstract}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Abstract}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Abstract}
   \@makeschapterhead{\centerline{Abstract}}
   \vskip 20pt\par\singlespace}{\newpage}%
 %% This version prints the thesis/dissertation title
 %% \renewenvironment{abstract}{\singlespace
-%%   \addcontentsline{toc}{chapter}{Abstract}
+%%   \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Abstract}
 %%   \@makeschapterhead{\centerline{Abstract}}
 %%   \vskip 20pt\par\noindent Title: \Ztitle\par
 %%   \vskip 0.5\baselineskip\strut\\%
@@ -452,8 +452,8 @@ Date Defended: \Zdefdate\\%
 % \iffalse
 %% \newenvironment{acknowledgments}{\singlespace\newpage
 \newenvironment{acknowledgments}{\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}Acknowledgments}
-  \addcontentsline{toc}{chapter}{Acknowledgments}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Acknowledgments}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Acknowledgments}
   \@makeschapterhead{\centerline{Acknowledgments}}
   \vskip 20pt\par}{\newpage}
 % \fi
@@ -464,7 +464,7 @@ Date Defended: \Zdefdate\\%
 % \iffalse
 \def\contentsname{Table of Contents}
 \renewcommand{\tableofcontents}{\singlespace\newpage
-  \addcontentsline{toc}{chapter}{Table of Contents}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Table of Contents}
   \@makeschapterhead{\centerline{Table of Contents}}
   \vskip 20pt\par\@starttoc{toc}}
 % \fi
@@ -474,8 +474,8 @@ Date Defended: \Zdefdate\\%
 % other document classes
 % \iffalse
 \renewcommand{\listoftables}{\singlespace\newpage
-\addcontentsline{toc}{chapter}{List of Tables}
-%%\addcontentsline{toc}{chapter}{\protect\numberline{}List of Tables}
+\cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Tables}
+%%\cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Tables}
   \@makeschapterhead{\centerline{List of Tables}}
   \vskip20pt
   \@starttoc{lot}}
@@ -486,8 +486,8 @@ Date Defended: \Zdefdate\\%
 % in other document classes
 % \iffalse
 \renewcommand{\listoffigures}{\singlespace\newpage
-  \addcontentsline{toc}{chapter}{List of Figures}
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}List of Figures}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Figures}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Figures}
   \@makeschapterhead{\centerline{List of Figures}}
   \vskip20pt
   \@starttoc{lof}}
@@ -496,12 +496,12 @@ Date Defended: \Zdefdate\\%
 % \DescribeEnv{listofsymbols} The |listofsymbols| environment lets you
 % include a table of symbols and acronyms.  You must build the table
 % yourself within the environment, using the table or other environments.
-% Use |\begin{listofsympols}| text 
+% Use |\begin{listofsympols}| text
 % |\end{listofsymols}| if you want one.
 % \iffalse
 \newenvironment{listofsymbols}{\singlespace\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}List of Symbols and Acronyms}
-  \addcontentsline{toc}{chapter}{List of Symbols and Acronyms}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Symbols and Acronyms}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Symbols and Acronyms}
   \@makeschapterhead{\centerline{List of Symbols and Acronyms}}
   \vskip 20pt\par}{\newpage}
 % \fi
@@ -512,32 +512,32 @@ Date Defended: \Zdefdate\\%
 % |\end{listofkeywords}| if you want one.
 % \iffalse
 \newenvironment{listofkeywords}{\singlespace\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}List of Keywords}
-  \addcontentsline{toc}{chapter}{List of Keywords}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Keywords}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Keywords}
   \@makeschapterhead{\centerline{List of Keywords}}
   \vskip 20pt\par}{\newpage}
 % \fi
 %
-% \DescribeEnv{dedication} The dedication environment 
+% \DescribeEnv{dedication} The dedication environment
 % creates a new page with at title for the dedication.  It
 % works very much like the |abstract| environment.
 % \iffalse
 %%\newenvironment{dedication}{\singlespace\newpage
 \newenvironment{dedication}{\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}Dedication}
-  \addcontentsline{toc}{chapter}{Dedication}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Dedication}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Dedication}
   \@makeschapterhead{\centerline{Dedication}}
   \vskip 20pt\par}{\newpage}
 % \fi
 %
-% \DescribeEnv{preface} The preface environment 
+% \DescribeEnv{preface} The preface environment
 % creates a new page with at title for the preface.  It
 % works very much like the |abstract| environment.
 % \iffalse
 %%\newenvironment{preface}{\singlespace\newpage
 \newenvironment{preface}{\newpage
-  %%  \addcontentsline{toc}{chapter}{\protect\numberline{}Preface}
-  \addcontentsline{toc}{chapter}{Preface}
+  %%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Preface}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Preface}
   \@makeschapterhead{\centerline{Preface}}
   \vskip 20pt\par}{\newpage}
 % \fi
@@ -562,8 +562,8 @@ Date Defended: \Zdefdate\\%
 \renewcommand{\thebibliography}[1]{
  \@makeschapterhead{\centerline{Bibliography}}
  \vskip20pt
-%% \addcontentsline{toc}{chapter}{\protect\numberline{}References}
- \addcontentsline{toc}{chapter}{Bibliography}
+%% \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}References}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Bibliography}
  \singlespace
  \list{[\arabic{enumi}]}{\settowidth\labelwidth{[#1]}
  \leftmargin\labelwidth 
@@ -585,7 +585,7 @@ Date Defended: \Zdefdate\\%
 % \iffalse
 \newenvironment{appendices}{
 \if@openright\cleardoublepage\else\clearpage\fi
- \addcontentsline{toc}{chapter}{Appendices}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Appendices}
  \thispagestyle{plain}\parindent\z@
  \parskip\z@ \@plus .3\p@\relax}
   {\clearpage}
@@ -602,12 +602,12 @@ Date Defended: \Zdefdate\\%
 \def\@appendix[#1]#2{\ifnum \c@secnumdepth >\m@ne
                          \refstepcounter{chapter}%
                          \typeout{\@chapapp\space\thechapter.}%
-                         \addcontentsline{toc}{section}%
+                         \cleardoublepage\phantomsection\addcontentsline{toc}{section}%
                          %%{\protect\numberline{Appendix \thechapter}}%
                          {Appendix \thechapter}%
                     \else
-                    %% \addcontentsline{toc}{section}{#1}%
-                    %% \addcontentsline{toc}{section}{}%
+                    %% \cleardoublepage\phantomsection\addcontentsline{toc}{section}{#1}%
+                    %% \cleardoublepage\phantomsection\addcontentsline{toc}{section}{}%
                     \fi
                     \chaptermark{#1}%
                     \addtocontents{lof}{\protect\addvspace{10\p@}}%
@@ -627,8 +627,8 @@ Date Defended: \Zdefdate\\%
 % \iffalse
 \newenvironment{gloss}{
  \newpage\@makeschapterhead{\centerline{Glossary of Terms}}
-%% \addcontentsline{toc}{chapter}{\protect\numberline{}Glossary of Terms}
- \addcontentsline{toc}{chapter}{Glossary of Terms}
+%% \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Glossary of Terms}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Glossary of Terms}
  \thispagestyle{plain}\parindent\z@
  \parskip\z@ \@plus .3\p@\relax}
  {\clearpage}
@@ -637,13 +637,13 @@ Date Defended: \Zdefdate\\%
 % \DescribeEnv{abbreviations} The |abbreviations| environment is used to create
 % list of abbreviations.  The environment starts a new page and puts an
 % appropriate heading, you have to fill in the text yourself.
-% Use |\begin{abbreviations}| |text| |\end{abbreviations}| to 
+% Use |\begin{abbreviations}| |text| |\end{abbreviations}| to
 % create a list of abbreviations.
 % \iffalse
 \newenvironment{abbreviations}{
  \@makeschapterhead{\centerline{List of Abbreviations}}
- %% \addcontentsline{toc}{chapter}{\protect\numberline{}List of Abbreviations}
- \addcontentsline{toc}{chapter}{List of Abbreviations}
+ %% \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}List of Abbreviations}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{List of Abbreviations}
  \thispagestyle{plain}\parindent\z@
  \parskip\z@ \@plus .3\p@\relax}
  {\clearpage}
@@ -669,8 +669,8 @@ Date Defended: \Zdefdate\\%
  \columnseprule \z@
  \columnsep 35\p@
  \twocolumn[\@makeschapterhead{\centerline{\indexname}}]%
-%% \addcontentsline{toc}{chapter}{\protect\numberline{}Index}
- \addcontentsline{toc}{chapter}{Index}
+%% \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Index}
+ \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Index}
  \thispagestyle{plain}\parindent\z@
  \parskip\z@ \@plus .3\p@\relax
  \let\item\@idxitem}
@@ -693,8 +693,8 @@ Date Defended: \Zdefdate\\%
 % \end{enumerate}
 % \iffalse
 \newenvironment{vita}{\newpage
-%%  \addcontentsline{toc}{chapter}{\protect\numberline{}Vita}
-  \addcontentsline{toc}{chapter}{Vita}
+%%  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{\protect\numberline{}Vita}
+  \cleardoublepage\phantomsection\addcontentsline{toc}{chapter}{Vita}
   \@makeschapterhead{\centerline{Vita}}
   \vskip 20pt\par}{\newpage}
 % \fi
-- 
2.20.1

